sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -2281,4 +2281,23 @@ class HiveDDLSuite
       }
     }
   }
+
+  test("SPARK-25301: checks view with udf from non default database") {
+    val db = "d1"
+    withDatabase(db) {
+      spark.sql(s"CREATE DATABASE $db")
+      val table = "t1"
+      withTable(s"$db.$table") {
+        spark.sql(s"CREATE TABLE $db.$table AS SELECT 'abcd' c1, 'xyz' c2")
+        val functionNameUpper = "udfUpper"
+        withUserDefinedFunction(s"$db.$functionNameUpper" -> false) {
+          val functionClass =
+            classOf[org.apache.hadoop.hive.ql.udf.generic.GenericUDFUpper].getCanonicalName
+          sql(s"CREATE FUNCTION $db.$functionNameUpper AS '$functionClass'")
+          val ds = sql(s"SELECT `$db.$functionNameUpper`(`$table`.`c1`)  FROM `$db`.`$table`")
```

Comment:
I think this is the expected behaivour of backquotes; ANTLR parses inputs like;
 - \`testdb.f1\` => `funcName`: f1 `dbName`: testdb
 - testdb.f1 => `funcName`: testdb.f1 `dbName`: default